### ISSUE TITLE
Swagger Model Validation flags readOnly properties as missing from request body example

### ISSUE THREAD

**Issue by mikekistler** [Azure/azure-sdk-tools#6840]
The Swagger Model Validation PR pipeline check is flagging as errors readOnly properties with `OBJECT_MISSING_REQUIRED_PROPERTY` because they are not present in the request body example.

An example is [PR 25444](https://github.com/Azure/azure-rest-api-specs/pull/25444):

<img width="1084" alt="image" src="https://github.com/Azure/azure-sdk-tools/assets/85643503/b75d6106-234f-43cf-9a56-13216c06e38c">

The problem likely stems from the (incorrect) assumption that `readOnly` could not be specified on `required` properties.

**Comment by mjmadhu** [comment]
@mikekistler @kurtzeborn we're blocked due to this issue and have our public preview SDK PLR pending sign-off as explained in the parent issue - https://github.com/Azure/typespec-azure/issues/3428. Could you please help with an ETA for resolution or any work-around for time-being?

**Comment by scbedd** [comment]
Hey apologies for the delay here. This is now merged.

[We needed to add a follow-up PR after enabling deep regression tests.](https://github.com/Azure/oav/pull/1019) There were some _insidious_ issues with the PR as-was, and I was really struggling for the free cycles to actually learn enough to understand what was going wrong.

We have also rolled out the change to `SwaggerValidation` and `ModelValidation`.
